### ISSUE TITLE
How to retrain my yoloV5s model

### ISSUE THREAD

**Issue by zain18jan2000** [ultralytics/yolov5#7456]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

I am trying to train my yoloVs model on colab, but maximum time for training is only 12 hours and my model 
need more than that (probably two or three days). So I want to train it in different intervals. How can I interupt and restart training of my model from where I left?

**Comment by glenn-jocher** [comment]
@zain18jan2000 👋 Hello! Thanks for asking about **resuming training**. YOLOv5 🚀 Learning Rate (LR) [schedulers](https://github.com/ultralytics/yolov5/blob/aa542ce6a65658ff931fee9bbab77c0145c152f0/train.py#L175-L181) follow predefined LR curves for the fixed number of `--epochs` defined at training start (`default=300`), and are designed to fall to a minimum LR on the final epoch for best training results. For this reason you can not modify the number of epochs once training has started.

<img width="850" alt="Screenshot 2022-04-10 at 11 11 51" src="https://user-images.githubusercontent.com/26833433/162611202-4e0b611f-6cca-4e43-94fc-bbc44e6447f8.png">

If your training was **interrupted** for any reason you may continue where you left off using the `--resume` argument. If your training fully completed, you can start a **new** training from any model using the `--weights` argument. Examples:

## Resume Single-GPU

You may not change settings when resuming, and **no additional arguments** other than `--resume` should be passed, with an optional path to the checkpoint you'd like to resume from. If no checkpoint is passed the most recently updated `last.pt` in your `yolov5/` directory is automatically found and used:
```bash
python train.py --resume  # automatically find latest checkpoint (searches yolov5/ directory)
python train.py --resume path/to/last.pt  # specify resume checkpoint
```

## Resume Multi-GPU

[Multi-GPU DDP trainings](https://docs.ultralytics.com/yolov5/tutorials/multi_gpu_training) must be resumed with the same GPUs and DDP command, i.e. assuming 8 GPUs:
```bash
python -m torch.distributed.run --nproc_per_node 8 train.py --resume  # resume latest checkpoint
python -m torch.distributed.run --nproc_per_node 8 train.py --resume path/to/last.pt  # specify resume checkpoint
```

## Start from Pretrained

If you would like to **start training** from a fully trained model, use the `--weights` argument, not the `--resume` argument:
```bash
python train.py --weights path/to/best.pt  # start from pretrained model
```

Good luck 🍀 and let us know if you have any other questions!

**Comment by merwynmohankumar** [comment]
Hi @glenn-jocher , Can we use the below code to retrain the best.pt model for new set of data. For example, we have trained a set of data in best.pt model . Now we have to use same best.pt model for new set of data. For this, how we can write the code

Start from Pretrained
If you would like to start training from a fully trained model, use the --weights argument, not the --resume argument:

python train.py --weights path/to/best.pt  # start from pretrained model

**Comment by glenn-jocher** [comment]
Yes

**Comment by merwynmohankumar** [comment]
Thanks @glenn-jocher ...Just to confirm the specific code..can I use below code to retrain the pretrained model for new set of data or do i need to make any changes in below code.?
" - python train.py --img 1000 --batch 8 --epochs 400 --data data.yaml --weights path/to/best.pt"

**Comment by glenn-jocher** [comment]
Sure, you can point to any —weights to use them as pretrained weights

**Comment by merwynmohankumar** [comment]
@glenn-jocher - Thanks for your comments

**Comment by glenn-jocher** [comment]
@merwynmohankumar you're welcome! If you have any other questions or need further assistance, feel free to ask. Good luck with your training!
